STM32F405RGTx/Core/Src/rtos_threads/state_machine_thread.cpp

```diff
@@ -124,7 +124,7 @@ State_t StateMachineThread::IdleEvent() {
 
     StateID requested_state;
     osStatus_t status = osMessageQueueGet(g_state_change_req_queue, &requested_state, NULL, 0);
-    if (status == osErrorTimeout) {
+    if (status == osErrorResource) {
```

Comment:
We need to fix this on the BMS state machine then as well, right?

Reply:
I don't think I used this strategy with the BMS. All that the CAN thread does in the BMS is send heartbeats and checks the flag for missed heartbeats (no StateID queue). The CAN thread for the MC is a bit more involved, just because it has to do more things over CAN (ex, manual control mode)